code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -359,6 +359,10 @@
 					if(isturf(S.loc) || ishuman(S.loc))
 						movement_target = S
 						break
+				for(var/obj/item/toy/rubber_ball/B in oview(src, 4))
```

Comment:
Honestly this should fall under wanted objects that I think @RemieRichards added but thats under hostile mob code so it'd be a pain to redo it for these guys to use it.

The best ways to code this are probably not worth the time spent for something that will never be used, so making it a generic pet plays with toy proc is probably the sanest way